Optimized Expression state controls, changed color variables

rlyriley opened the pull request:
Also some miscellaneous changes to the keyframe system, I suggest not merging those

rlyriley committed:
Optimized Expression state controls

rlyriley committed:
Changed colors

Changed from many unsigned ints to few unsigned int 2d arrays

rlyriley committed:
Changed gfx_pc.c to work with new colors

Technically it shouldn't work, but it does. I will not question the dark magic that's holding this up

rlyriley committed:
Changed colors to use array instead of many ints

rlyriley committed:
Updated colors.h
run_cc_replacement function now takes 3 ints, variable address changed from std::string to int

rlyriley committed:
Final adjustement to colors.cpp
I physically cannot look at colors.cpp anymore

rlyriley committed:
Final saturn_colors.cpp optimization[#2]
I think I'm addicted.

rlyriley committed:
saturn_colors.cpp optimizations #3
Fixed issues:
https://github.com/Yegg12/Saturn/issues/1
https://github.com/Yegg12/Saturn/issues/2

Llennpie commented:
These are nice optimizations but they break in February's update commits (mostly just changes relating to the work-in-progress keyframe system left in the code) ... Mind resolving the conflicts so I can merge?

rlyriley committed:
Merge branch 'legacy' into legacy

rlyriley committed:
Fixed 2D array in gfx_pc.c,
reverted address variable in saturn_colors.cpp back to int

rlyriley committed:
Controllers work while window is unfocused

rlyriley committed:
body parts no longer arrays, now structs

rlyriley commented:
Optimizations done, ready for pull

Llennpie committed:
Minor bug fixes & touch-ups

PR changes:
- Fixed some color parts not updating correctly
- Moved SDL_SetHint to a more convenient location
- Changed UI appearance (I'm nitpicky)
